examples/run_parallel_and_cleanup.py

```diff
@@ -0,0 +1,86 @@
+"""Script to run all example files"""
+import os
+import subprocess
+import shutil
+import sys
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+#checks if we are in the examples directory
+if not os.path.basename(os.getcwd()) == "examples":
+    print("Run this script from the examples directory.")
+    sys.exit(1)
+# Parse command-line arguments
+if len(sys.argv) > 2:
+    print("Usage: python run_parallel_and_cleanup.py [max_threads]")
+    sys.exit(1)
+
+# Default to 2 threads if no argument is provided
+max_threads = int(sys.argv[1]) if len(sys.argv) == 2 else 4
+if max_threads <= 0:
+    print("Maximum number of threads must be a positive integer.")
+    sys.exit(1)
+
+# Define the log directory
+log_dir = "logs"
+os.makedirs(log_dir, exist_ok=True)
+
+
+# Function to run a Python script and log its output
+def run_script(file_path):
+    script_name = os.path.basename(file_path)
+    log_file = os.path.join(log_dir, f"{script_name}.log")
+    print(f"Running {file_path}...")
+
+    try:
+        # Run the script and capture stdout/stderr
+        result = subprocess.run(["python3", file_path],
+                                capture_output=True,
+                                text=True,
+                                check=True)
+        # Write logs to the log file
+        with open(log_file, "w", encoding="utf-8") as log:
+            log.write(f"--- Output for {script_name} ---\n")
+            log.write("STDOUT:\n")
+            log.write(result.stdout + "\n")
+            log.write("STDERR:\n")
+            log.write(result.stderr + "\n")
+        print(f"Logs for {script_name} saved to {log_file}")
+    except subprocess.CalledProcessError as e:
+        # Handle errors and write to log
+        with open(log_file, "w", encoding="utf-8") as log:
+            log.write(f"--- Error running {script_name} ---\n")
+            log.write("STDOUT:\n")
+            log.write(e.stdout + "\n")
+            log.write("STDERR:\n")
+            log.write(e.stderr + "\n")
+        print(f"Error logs for {script_name} saved to {log_file}")
+
+
+# Step 1: Find all Python files, excluding this script
```

Comment:
I know that this is really just a simple "script" to run the examples, but please try to avoid this anti-pattern:
https://inductiva.ai/blog/article/programming-at-inductiva#dont-write-scripts

And at a minimum, define small functions for each of the sections you are separating with comment headers.

Reply:
Done

Reply:
Thanks Paulo!